app/streaming/video/ffmpeg-renderers/d3d11va.cpp

```diff
@@ -698,14 +1239,189 @@ void D3D11VARenderer::renderVideo(AVFrame* frame)
+
+    ZeroMemory(&m_OutputViewDesc, sizeof(m_OutputViewDesc));
+    m_OutputViewDesc.ViewDimension = D3D11_VPOV_DIMENSION_TEXTURE2D;
+    m_OutputViewDesc.Texture2D.MipSlice = 0;
+
+    hr = m_VideoDevice->CreateVideoProcessorOutputView(
+        m_BackBufferResource,
+        m_VideoProcessorEnumerator,
+        &m_OutputViewDesc,
+        (ID3D11VideoProcessorOutputView**)&m_OutputView);
+    if (FAILED(hr)) {
+        return false;
+    }
+
+    ZeroMemory(&m_InputViewDesc, sizeof(m_InputViewDesc));
+    m_InputViewDesc.FourCC = 0;
+    m_InputViewDesc.ViewDimension = D3D11_VPIV_DIMENSION_TEXTURE2D;
+    m_InputViewDesc.Texture2D.MipSlice = 0;
+    m_InputViewDesc.Texture2D.ArraySlice = 0;
+
+    hr = m_VideoDevice->CreateVideoProcessorInputView(
+        m_VideoTexture, m_VideoProcessorEnumerator, &m_InputViewDesc, (ID3D11VideoProcessorInputView**)&m_InputView);
+    if (FAILED(hr))
+        return false;
+
+    // Apply processed filters to the surface
+    RECT srcRect = { 0 };
+    srcRect.right = m_DecoderParams.width;
+    srcRect.bottom = m_DecoderParams.height;
+
+    RECT dstRect = { 0 };
+    dstRect.right = m_DisplayWidth;
+    dstRect.bottom = m_DisplayHeight;
+
+    // Sscale the source to the destination surface while keeping the same ratio
+    float ratioWidth = static_cast<float>(m_DisplayWidth) / static_cast<float>(m_DecoderParams.width);
+    float ratioHeight = static_cast<float>(m_DisplayHeight) / static_cast<float>(m_DecoderParams.height);
+
+    // [TODO] There is a behavior I don't understand (bug?) when the destination desRect is larger by one of its side than the source srcRect.
+    // If it is bigger, the window becomes black, but if it is smaller it is fine.
+    // Only one case is working when it is bigger is when the dstRest perfectly equal to the Display size.
+    // Investigation: If there anything to do with pixel alignment (c.f. dxva2.cpp FFALIGN), or screenSpaceToNormalizedDeviceCoords ?
+    // Fix: When bigger we strech the picture to the window, it will be deformed, but at least will not crash.
+    if(m_DisplayWidth < m_DecoderParams.width && m_DisplayHeight < m_DecoderParams.height){
+        if(ratioHeight < ratioWidth){
+            // Adjust the Width
+            long width = static_cast<long>(std::floor(m_DecoderParams.width * ratioHeight));
+            dstRect.left = static_cast<long>(std::floor(  abs(m_DisplayWidth - width) / 2  ));
+            dstRect.right = dstRect.left + width;
+        } else if(ratioWidth < ratioHeight) {
+            // Adjust the Height
+            long height = static_cast<long>(std::floor(m_DecoderParams.height * ratioWidth));
+            dstRect.top = static_cast<long>(std::floor(  abs(m_DisplayHeight - height) / 2  ));
+            dstRect.bottom = dstRect.top + height;
+        }
+    }
+
+    m_VideoContext->VideoProcessorSetStreamSourceRect(m_VideoProcessor, 0, true, &srcRect);
```

Comment:
I don't think source rect needs to be set because it reads from the entire surface by default

Reply:
Done
Correct, it works without setting this value.